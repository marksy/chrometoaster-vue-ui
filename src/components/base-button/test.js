import Vue from 'vue';
import { shallow } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';
import BaseButton from './';

Vue.config.productionTip = false

describe('<base-button />', () => {
  const Component = Vue.extend(BaseButton)
  it('does not crash', () => {
    const vm = new Component().$mount()
    expect(vm.$el.textContent).toMatch(/BaseButton/)
  });
  it('allows props to be passed in', () => {
    const vm = new Component({ propsData: {
      className: 'base-button--alt'
    }}).$mount();
    expect(vm.$el.classList).toContain('base-button--alt');
  });
  it('matches wrapper snapshot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(Component);
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
