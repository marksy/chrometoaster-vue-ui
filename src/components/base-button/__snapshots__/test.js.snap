// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<base-button /> matches wrapper snapshot 1`] = `"<button data-server-rendered=\\"true\\" class=\\"base-button\\">BaseButton</button>"`;
